server/service/osquery_utils/queries.go

```diff
@@ -1201,8 +1201,9 @@ func directIngestMunkiInfo(ctx context.Context, logger log.Logger, host *fleet.H
 
 func ingestKubequeryInfo(ctx context.Context, logger log.Logger, host *fleet.Host, rows []map[string]string) error {
 	if len(rows) != 1 {
-		logger.Log("component", "service", "method", "ingestKubequeryInfo", "warn",
-			fmt.Sprintf("kubernetes_info expected single result got %d", len(rows)))
+		err := fmt.Errorf("kubernetes_info expected single result got: %d", len(rows))
+		logger.Log("component", "service", "method", "ingestKubequeryInfo", "warn", err)
+		return err
```

Comment:
If we are warning, should we return an err here? It'd be logged twice, once here and another in the ingestion bit.

Reply:
True, we are logging twice in many places.

Reply:
[33410ff](https://github.com/fleetdm/fleet/pull/8178/commits/33410ffb46615213ab7bf86a434197d8e5a98047).